sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -930,7 +931,7 @@ private <T> Flux<FeedResponse<T>> createQuery(
 
         return ObservableHelper.fluxInlineIfPossibleAsObs(
             () -> createQueryInternal(
-                resourceLink, sqlQuery, options, klass, resourceTypeEnum, queryClient, correlationActivityId),
+                resourceLink, sqlQuery, options, klass, resourceTypeEnum, queryClient, correlationActivityId, isQueryCancelledOnTimeout),
```

Comment:
maybe isQueryCancelledOnTimeout can also be part of CosmosQueryRequestOptions. We have been used it for a lot other tracking purpose

Reply:
Discussed offline - `CosmosQueryRequestOptions` can maintain state across various query operations and `isQueryCancelledOnTimeout` should not be shared across queries. Also, the SDK only creates a shallow copy of `CosmosQueryRequestOptions`.